Added week 1 & 2 homework

Catsudemo opened the pull request:
Week 2 is functional but could use some tidying

Catsudemo committed:
Added week 1 & 2 homework

Week 2 is functional but could use some tidying

marianabocoi commented on week1/homework/src/commands.js:
```diff
@@ -0,0 +1,25 @@
+      return state;
+    default:
+      return state;
+
```
remove empty line

marianabocoi commented on week1/homework/src/package.json:
```diff
@@ -0,0 +1,15 @@
+{
```
the homework project already has a `package.json`, no need to create another one.

marianabocoi replied:
ping @Catsudemo

marianabocoi commented on week1/homework/src/responses/sendIndexPage.js:
```diff
@@ -0,0 +1,17 @@
+function sendIndexPage(response) {
```
I am not sure creating a landing page was part of the requirements for this homework.  Please remove.

marianabocoi replied:
Please remove page

marianabocoi commented on week1/homework/src/responses/sendOtherPage.js:
```diff
@@ -0,0 +1,19 @@
+'use strict';
```
I am not sure that was part of the requirements either. please remove.

marianabocoi commented on week1/homework/src/commands.js:
```diff
@@ -0,0 +1,25 @@
+
```
remove empty line or add `use strict` on top ?

marianabocoi commented on week1/homework/src/responses/sendStyles.js:
```diff
@@ -0,0 +1,13 @@
+'use strict';
```
this looks like copy-paste without much cleanup. Please remove.

marianabocoi commented on week1/homework/src/commands.js:
```diff
@@ -0,0 +1,25 @@
+
+function whatURL(command, state) {
```
This function name is slightly confusing especially since it returns state.  What would be a better name?

marianabocoi commented on week1/homework/src/commands.js:
```diff
@@ -0,0 +1,25 @@
+  }
+}
+
+module.exports = whatURL
```
I like that you exported this in a separate module that handles commands.

marianabocoi commented on week1/homework/src/server.js:
```diff
@@ -1,16 +1,37 @@
 'use strict';
 
 const http = require('http');
+const whatIDO = require('./commands')
```
What would be a better name for this variable?

marianabocoi commented on week1/homework/src/server.js:
```diff
@@ -1,16 +1,37 @@
+
+
+
+
```
remove many empty lines.

marianabocoi commented on week1/homework/src/server.js:
```diff
@@ -1,16 +1,37 @@
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+    const URLString = request.url;
+    const URLArray = URLString.split('/').filter(command => ["state", "add", "subtract", "reset"].includes(command));
```
you do not really need to split. You will never get concatenated commands, it was a misunderstanding when reading the test output.

marianabocoi commented on week1/homework/src/server.js:
```diff
@@ -1,16 +1,37 @@
+      state = whatIDO(command, state)
+    }
+
+    response.writeHead(200, { 'Content-Type': 'application/json' });
```
the header should be 403 if the path is not the specified one.

apooslr replied:
404 is maybe better ?

marianabocoi commented on week2/homework/src/todolist.txt:
```diff
@@ -0,0 +1,8 @@
+
```
This should not be added to git. Add this file to the `.gitignore` file.

marianabocoi commented on week1/homework/src/responses/sendText.js:
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+function sendText(response, text) {
```
should it set also the status code?

marianabocoi reviewed (changes requested):
Week1. I can see where you are going but it needs a bit of cleanup. 💃 
- Copy-paste files that are not used and lingered. 
- All tests fail - please fix
- Formatting errors
- Supporting paths like "/add/add/subtract" that are not needed -> partially my fault, sorry. Please simplify down.

marianabocoi commented on week2/homework/src/helpfile.txt:
```diff
@@ -0,0 +1,6 @@
+Add - Type add plus the reminder you want to add to your list. E.g. add buy milk. 
+Remove - Type remove plus the number of the reminder you want to remove from your list. E.g. remove 2.
+List - Type list to read everything currently on your list. 
+Reset - Type reset to remove everything from your list. Don't worry, you'll be asked to confirm before resetting. 
```
commands should probably be lowercase.

marianabocoi commented on week2/homework/src/index.js:
```diff
@@ -1,3 +1,59 @@
+}
+
+
+
```
cleanup newlines

marianabocoi commented on week2/homework/src/index.js:
```diff
@@ -1,3 +1,59 @@
+
+
+
+function whichDo(file, list, theFunction, input) {
```
there is probably a more readable name out there. Which would you say that is?

apooslr replied:
There is probably a better way than doing `else if` so many times

marianabocoi commented on week2/homework/src/index.js:
```diff
@@ -1,3 +1,59 @@
+
+function whichDo(file, list, theFunction, input) {
+  if (theFunction == 'add') {
+    addToList(file, list, input)
```
check if `input` is Undefined.

marianabocoi commented on week2/homework/src/modules/removeFromList.js:
```diff
@@ -0,0 +1,6 @@
+function removeFromList(list, input) {
+  list.splice(input, 1)
```
is there a simpler way of doing this?

marianabocoi replied:
You should shift it so it removes 1-base indexes

marianabocoi commented on week2/homework/src/modules/resetList.js:
```diff
@@ -0,0 +1,8 @@
+  let all = list.length;
+  list.splice(0, all);
+  console.log('Okay, everything has been removed from your list. ')
+  return list
```
can be simpler. how can it be simpler?

marianabocoi commented on week2/homework/src/modules/confirmReset.js:
```diff
@@ -0,0 +1,11 @@
+function confirmReset(list) {
+  if (list.length > 0) {
```
this is not exactly working as it should.

marianabocoi reviewed (changes requested):
Week2: Some functionality is missing. 
- list - works 👍  - nitpick you can list as a list with index? 
- add - works 👍 - should probably check if there is an input
- remove - works but the index it removes is off by 1. It should remove a to-do item by its 1-base index
- reset - does not work

apooslr commented on week1/homework/src/commands.js:
```diff
@@ -0,0 +1,25 @@
+function whatURL(command, state) {
+  switch (command) {
+    case 'state':
+      console.log('checked state')
```
console.log are fine to use for cli applications, for when we really want to be informative, for example in the server when we say what port the server is listening on etc. If we use it for debugging purposes the prints should be removed when we are done debugging.

apooslr commented on week2/homework/src/index.js:
```diff
@@ -1,3 +1,59 @@
+var fs = require('fs')
+let filename = './todolist.txt';
+let currentList = fs.readFileSync(filename, "utf-8").toString().split("\n");
+let whichFunction = process.argv[2];
```
is it really a function that we are receiving ?

apooslr commented on week2/homework/src/package.json:
```diff
@@ -0,0 +1,12 @@
+  "version": "1.0.0",
+  "description": "CLI-app-ToDoList",
+  "main": "index.js",
+  "start: "index.js"
```
I dont think this works

AndreaRylander commented on week2/homework/src/index.js:
```diff
@@ -1,3 +1,59 @@
 
-// TODO: Write the homework code in this file
+var fs = require('fs')
+let filename = './todolist.txt';
```
What would happen if this file doesnt exist?

apooslr commented on week2/homework/src/index.js:
```diff
@@ -1,3 +1,59 @@
-// TODO: Write the homework code in this file
+var fs = require('fs')
+let filename = './todolist.txt';
+let currentList = fs.readFileSync(filename, "utf-8").toString().split("\n");
```
What would happen if this file doesnt exist?

apooslr commented on week2/homework/src/index.js:
```diff
@@ -1,3 +1,59 @@
+let currentList = fs.readFileSync(filename, "utf-8").toString().split("\n");
+let whichFunction = process.argv[2];
+let rawInput = process.argv.splice(3).join(' ')
+let userInput = `${rawInput}`
```
The `rawInput` variable declared on line 7 is only used on line 8. Why do we need to declare this variable at all?

apooslr reviewed (changes requested):
Week 1:
- 
Breaking out logic is good but consider when it makes sense to do it and when it doesnt. 
Always try to have  names for functions and variables that are descriptive, it makes your code easier to read and look more professional over all.
As Mariana mentioned,  all the test are failing, please address that.

Week 2:
-
* When the `todolist.txt` file is empty, the  command `list` show an list with an empty string, it would be more accurate if it showed an empty list. 
* The reset command doesnt really work. It asks the user to type confirm, to reset the list. But actually you mean to ask the user to use the command `confirm`. But consider the specifications for the command `reset`. 
* Naming is important, consider colleagues or yourself having to go back and look at the code in a few weeks or months, would they understand the names of functions or variables? 
* Remove the empty `whichDo.js`file

Catsudemo committed:
Updated week 2

Fixed offset by 1 for remove.
List now lists item's number on list. - created a new function/module to handle listing.
Empty list now no longer returns an empty string when listing.
Removed unneeded file.

Catsudemo committed:
Fixed problem with writing new file

App can now run without a pre-existing text file.

Catsudemo committed:
Updated to handle errors properly

Catsudemo committed:
Added week 3 homework

To Do App with additional functions

Catsudemo committed:
Fixed week 2

Fixed off by one.
Not sure what wasn't working about reset but I removed my nice confirmation function anyway.

Catsudemo committed:
Swapped put for post

Swapped put for post for updating and marking as done.

Catsudemo committed:
Update helpfile.txt

Removed confirm from the helpfule

Catsudemo committed:
Updated week 1

Cleaned up some empty lines.

Catsudemo committed:
Fixed week 1

Week 1 now passing tests.

marianabocoi reviewed (changes requested):
Week1:
Tests pass!!
remove:
week1/homework/src/package.json (there is already a package.jsone one level higher)
week1/homework/src/responses/ folder (not used)
week1/homework/src/package-lock.json (generated by the wrong package.json)

Week2
All works
remove
week2/homework/src/package.json
